ledger/src/blockstore.rs

```diff
@@ -3808,21 +3846,44 @@ where
     Ok(())
 }
 
+// Update the meta for a slot whose parent is known to be connected.
+// Returns a bool indicating whether this slot is now connected which would
+// indicate that it's children are also in need of update.
+fn update_child_from_connected_parent(meta: &mut SlotMeta) -> bool {
+    // If this slot is already connected, its' children should
+    // also already be connected so cut the traversal here.
```

Comment:
Technically we could, but we would need to do so recursively.
- check the children
- check the children's children
- ...

Given that this function is only called at startup (and practically speaking, it should only traverse a couple hundred slots if people are using sane snapshot settings), doing the full traversal here wouldn't be terrible.

However, this PR's behavior for setting a slot to be connected is pretty tight so given that, I'm inclined to think that an assertion / double check is overkill. But, if the assumption is somehow not true, that would break the functionality of the connected flag until the next load from snapshot. So, maybe worth doing the traversal and setting flags again just incase?